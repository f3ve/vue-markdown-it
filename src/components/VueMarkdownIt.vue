<script lang="ts" setup>
import MarkdownIt from 'markdown-it';

const props = withDefaults(
  defineProps<{
    /** Markdown text to render */
    source: string;
    /**
     * Custom class name to apply to markdown container
     * @default 'vue-md-it-wrapper'
     */
    mdWrapperClass?: string;
    plugins?: (
      | MarkdownIt.PluginSimple
      | [MarkdownIt.PluginSimple | MarkdownIt.PluginWithOptions<any>, any]
    )[];
    preset?: MarkdownIt.PresetName;
    options?: MarkdownIt.Options;
  }>(),
  {
    mdWrapperClass: 'vue-md-it-wrapper',
  },
);

const md = props.preset
  ? new MarkdownIt(props.preset, props.options)
  : props.options
    ? new MarkdownIt(props.options)
    : new MarkdownIt();

if (props.plugins) registerPlugins();

function registerPlugins() {
  props.plugins?.forEach((item) => {
    const plugin = toArray(item);
    md.use(plugin[0], plugin[1]);
  });
}

function toArray<T = any>(val: T) {
  return Array.isArray(val) ? val : [val, undefined];
}
</script>

<template>
  <div :class="mdWrapperClass" v-html="md.render(source)"></div>
</template>
