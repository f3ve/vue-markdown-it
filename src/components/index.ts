export { default as VueMarkdownIt } from './VueMarkdownIt.vue';
