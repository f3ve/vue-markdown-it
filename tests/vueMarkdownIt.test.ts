import { mount } from '@vue/test-utils';
import { VueMarkdownIt } from '@/components';

test('Mount VueMarkdownIt', async () => {
  expect(VueMarkdownIt).toBeTruthy();

  const wrapper = mount(VueMarkdownIt, {
    props: {
      source: `# This is a test!
      Muffins are awesome
      `,
    },
  });

  console.log(wrapper.html());

  expect(wrapper.html()).toContain('<h1>This is a test!</h1>');
  expect(wrapper.html()).toMatchSnapshot();
});
