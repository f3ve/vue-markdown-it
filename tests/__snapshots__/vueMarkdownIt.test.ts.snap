// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount VueMarkdownIt 1`] = `
"<div class=\\"vue-md-it-wrapper\\">
  <h1>This is a test!</h1>
  <pre><code>  Muffins are awesome
</code></pre>
</div>"
`;
